s3stream/src/main/java/com/automq/stream/s3/wal/impl/block/SlidingWindowService.java

```diff
@@ -331,23 +331,10 @@ private BlockBatch pollBlocksLocked() {
      * Finish the given block batch, and return the start offset of the first block which has not been flushed yet.
      */
     private long wroteBlocks(BlockBatch wroteBlocks) {
-        blockLock.lock();
-        try {
-            return wroteBlocksLocked(wroteBlocks);
-        } finally {
-            blockLock.unlock();
-        }
-    }
-
-    /**
-     * Finish the given block batch, and return the start offset of the first block which has not been flushed yet.
-     * Note: this method is NOT thread safe, and it should be called with {@link #blockLock} locked.
-     */
-    private long wroteBlocksLocked(BlockBatch wroteBlocks) {
         boolean removed = writingBlocks.remove(wroteBlocks.startOffset());
         assert removed;
         if (writingBlocks.isEmpty()) {
-            return getCurrentBlockLocked().startOffset();
+            return wroteBlocks.startOffset() + WALUtil.alignLargeByBlockSize(wroteBlocks.blockBatchSize());
```

Comment:
I think there is an issue here.

The comments in the `wroteBlocks` method state that it "returns the start offset of the first block which has not been flushed yet". However, `wroteBlocks.endOffset()` is not necessarily a block that has not been flushed yet.

For example, suppose there are initially 3 blocks. Block-1{start=10, end=20}, Block-2{start=20, end=30}, Block-3{start=30, end=40}. And they call `wroteBlocks` in the order Block-1 -> Block-3 -> Block-2.
1. `wroteBlocks(Block-1)` returns 20 (the start offset of Block-2)
2. `wroteBlocks(Block-3)` returns 20 too, as the Block-2 has not been flushed yet.
3. `wroteBlocks(Block-2)` will return 30 (the end offset of Block-2). However, 40 (the end offset of Block-3) is expected, as the Block-3 has already been flushed.